lib/ramble/ramble/application_types/spack.py

```diff
@@ -365,6 +365,21 @@ def _mirror_software(self, workspace):
             with self.logger.force_echo():
                 tty.die(e)
 
+    register_phase('push_to_spack_cache', pipeline='pushtocache', depends_on=[])
+
+    def _push_to_spack_cache(self, workspace):
+        try:
+            # TODO: make sure this is called as infrequently as possible
+            # TODO: this is a lot of repeated stuff, can we DRY with out pipelines?
```

Comment:
It is called a lot of times. Maybe there's a better way we can do it though.
I don't have any great suggestions on where this should go though. Maybe a wrapper within the runner?

You probably also want to make sure you deactivate the spack_runner.

Reply:
Good catch on the deactivate.

I added an `if already_pushed` to the spack_runner which will help some